Tests/Form/FormMapperTest.php

```diff
@@ -429,6 +429,20 @@ public function testKeys()
         $this->assertSame(array('foo', 'baz'), $this->formMapper->keys());
     }
 
+    public function testFieldNameIsSanitized()
+    {
+        $this->contractor->expects($this->any())
+            ->method('getDefaultOptions')
+            ->will($this->returnValue(array()));
+
+        $this->formMapper
+            ->add('fo.o', 'bar')
+            ->add('ba__z', 'foobaz')
+        ;
+
+        $this->assertSame(array('fo__o', 'ba____z'), $this->formMapper->keys());
```

Comment:
> the sanitize method has to be reversible.

Yes it has to. Sadly, it is in only reversed in the `FormMapper` but not in the `FormBuilder`. So calling `FormBuilder::get('ba__z')` (in a Form Event for example) throws an exception. And it was not the case before this PR. This is a BC break.